Formula/r.rb

```diff
@@ -56,6 +57,9 @@ def install
       ENV.append "LDFLAGS", "-L#{Formula[f].opt_lib}"
     end
 
+    # Enable binaries on macos
+    ENV.append "CPPFLAGS", "-DPLATFORM_PKGTYPE='\"mac.binary\"'"
```

Comment:
Thanks for your PR, @jonkeane.
```suggestion
    on_macos do
      # Enable binaries on macos
      ENV.append "CPPFLAGS", "-DPLATFORM_PKGTYPE='\"mac.binary\"'"
    end
```
Let's wrap this in an `on_macos` block so that it doesn't cause problems for Linux users.

Reply:
Oh, and please squash this change with your first commit when you do it. Thanks!

Reply:
I can absolutely add that, funny I actually had that in my first draft, but the `audit` told me to get rid of it so I did. No big deal to add it back. I also suspect that it would be good to _not_ enable this on arm64_big_sur. would something like the following be the right way to detect that the formula was being built on Apple silicon? 

```
on_macos do
      # Enable binaries on macos
      ENV.append "CPPFLAGS", "-DPLATFORM_PKGTYPE='\"mac.binary\"'" unless Hardware::CPU.arm?
end
```

Reply:
Yes, that's correct. I also suspect you have too many quotes up there, and that you don't actually need any of them.

Reply:
Ok, I just sent an updated version. I tried it without the quotes (as well as with single versions of both) and the make step failed — it looks like https://github.com/wch/r-source/blob/74bb4f560ee02070ae631e1ca66ef6a16f256b24/src/main/platform.c#L303 is expecting that (odd) set of quoting to work.